async/cohttp_async.ml

```diff
@@ -25,14 +25,14 @@ module Net = struct
   let lookup uri =
     let host = Option.value (Uri.host uri) ~default:"localhost" in
     match Uri_services.tcp_port_of_uri ~default:"http" uri with
-    | None -> raise (Failure "Net.connect") (* TODO proper exception *)
+    | None -> failwith "Net.connect" (* TODO proper exception *)
     | Some port ->
       let open Unix in
       Addr_info.get ~host [Addr_info.AI_FAMILY PF_INET; Addr_info.AI_SOCKTYPE SOCK_STREAM]
       >>= function
       | {Addr_info.ai_addr=ADDR_INET(addr,_)}::_ ->
         return (host, Ipaddr_unix.of_inet_addr addr, port)
-      | _ -> raise (Failure "resolution failed")
+      | _ -> failwith "resolution failed"
```

Comment:
Yes all of this stuff is very much less than ideal. I think we are saving the refactoring post 1.0 however.